File: app/src/test/java/com/google/android/stardroid/units/LatLongTest.java
package com.google.android.stardroid.units;

import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;

/**
 * Created by johntaylor on 2/15/16.
 */
public class LatLongTest {
    private static final float TOL = 1e-5f;

    @Test
    public void latLong_testInstantiatesCorrectly() {
        LatLong position = new LatLong(45, 50);
        assertThat(position.getLatitude()).isWithin(TOL).of(45f);
    }

    @Test
    public void latLong_testBoundsCorrectly() {
        LatLong position = new LatLong(95, 50);
        assertThat(position.getLatitude()).isWithin(TOL).of(90f);

        position = new LatLong(-105, 50);
        assertThat(position.getLatitude()).isWithin(TOL).of(-90f);

        position = new LatLong(45, 240);
        assertThat(position.getLongitude()).isWithin(TOL).of(-120f);

        position = new LatLong(45, -200);
        assertThat(position.getLongitude()).isWithin(TOL).of(160f);

        position = new LatLong(45, 600);
        assertThat(position.getLongitude()).isWithin(TOL).of(-120f);

        position = new LatLong(45, -560);
        assertThat(position.getLongitude()).isWithin(TOL).of(160f);
    }
}

File: app/src/test/java/com/google/android/stardroid/units/LatLongTest.kt
package com.google.android.stardroid.units

import com.google.common.truth.Truth.assertThat
import org.junit.Test

/**
 * Created by johntaylor on 2/15/16.
 */
class LatLongTest {
    @Test
    fun latLong_testInstantiatesCorrectly() {
        val position = LatLong(45f, 50f)
        assertThat(position.latitude).isWithin(TOL).of(45f)
    }

    @Test
    fun latLong_testBoundsCorrectly() {
        var position = LatLong(95f, 50f)
        assertThat(position.latitude).isWithin(TOL).of(90f)
        position = LatLong(-105f, 50f)
        assertThat(position.latitude).isWithin(TOL).of(-90f)
        position = LatLong(45f, 240f)
        assertThat(position.longitude).isWithin(TOL).of(-120f)
        position = LatLong(45f, -200f)
        assertThat(position.longitude).isWithin(TOL).of(160f)
        position = LatLong(45f, 600f)
        assertThat(position.longitude).isWithin(TOL).of(-120f)
        position = LatLong(45f, -560f)
        assertThat(position.longitude).isWithin(TOL).of(160f)
    }

    companion object {
        private const val TOL = 1e-5f
    }
}